CppCoreGuidelines.md

```diff
@@ -7296,15 +7296,57 @@ Casting to a reference expresses that you intend to end up with a valid object,
 
 ##### Reason
 
-???
+A failure to find the required class will cause `dynamic_cast` to return a null value, and de-referencing a null-valued pointer will lead to undefined behaviour.
+Therefore the result of the `dynamic_cast` should always be treated as if it may contain a null value, and tested.
```

Comment:
These sentences are 100% true, but they describe how to avoid undefined behaviour when following this guideline, they don't actually give a reason for the guideline itself.

I would expect the rationale to start with something explaining that when it is normal expected behaviour for the cast to fail, you don't want to have to deal with an exception being thrown. So casting to a pointer and checking the result is the right approach. _Then_ your text explaining that testing the result can't be omitted.